### PR TITLE
Assert on access to unk_982134

### PR DESCRIPTION
**Pull request by janisozaur**
Introduced by #4100

### CODE REVIEW

**Comment by janisozaur** [comment]
This gets hit on the very first park in default title sequence for me

**Comment by janisozaur** [comment]
Backtrace:
```
#5  0x01434295 in set_sprite_type (peep=0x111a1bc, type=25 '\031') at /home/janisozaur/workspace/OpenRCT2/src/peep/peep.c:1294
#6  0x01434187 in peep_update_sprite_type (peep=0x111a1bc) at /home/janisozaur/workspace/OpenRCT2/src/peep/peep.c:1442
#7  0x014327b8 in sub_68F41A (peep=0x111a1bc, index=716) at /home/janisozaur/workspace/OpenRCT2/src/peep/peep.c:658
#8  0x01431ff6 in peep_update_all () at /home/janisozaur/workspace/OpenRCT2/src/peep/peep.c:382
#9  0x0142caee in game_logic_update () at /home/janisozaur/workspace/OpenRCT2/src/game.c:368
#10 0x01457afb in title_update () at /home/janisozaur/workspace/OpenRCT2/src/title.c:528
#11 0x0145b012 in rct2_update () at /home/janisozaur/workspace/OpenRCT2/src/rct2.c:456
#12 0x015e4038 in openrct2_loop () at /home/janisozaur/workspace/OpenRCT2/src/openrct2.c:395
#13 0x015e34a0 in openrct2_launch () at /home/janisozaur/workspace/OpenRCT2/src/openrct2.c:324
#14 0x015de34d in main (argc=1, argv=0xffffd634) at /home/janisozaur/workspace/OpenRCT2/src/platform/posix.c:60
```

**Comment by janisozaur** [comment]
This is getting called from https://github.com/OpenRCT2/OpenRCT2/blob/964cf66f6b4860d518ed67d041bda222de5f29ca/src/peep/peep.c#L1441 
```C
set_sprite_type(peep, PEEP_SPRITE_TYPE_25);
```

`PEEP_SPRITE_TYPE_25` is defined to equal 25 in here: https://github.com/OpenRCT2/OpenRCT2/blob/964cf66f6b4860d518ed67d041bda222de5f29ca/src/peep/peep.h#L350

`unk_982134` is only 16 entries long in here: https://github.com/OpenRCT2/OpenRCT2/blob/964cf66f6b4860d518ed67d041bda222de5f29ca/src/peep/peep.c#L335-L337

**Comment by duncanspumpkin** [comment]
Yeah this is caused by CS being sneaky. 0x00982144 starts at the end of this array (at first glance). But really this address is used by high shop item numbers i.e. >32 so really 0x00982144 is not part of this different array. unk_982134 should really be 44 elements long.

**Comment by duncanspumpkin** [comment]
Replace unk_982134 with the following information 
```00 00 00 00|00 00 00 00|00 00 00 01|00 00 01 01 
01 01 01 00|01 00 01 01|01 00 00 01|01 00 00 01 
01 01 01 01|01 01 00 01|00 01 01 01|01 01 01 01```

**Comment by janisozaur** [comment]
I've added missing data now. @duncanspumpkin can you verify/merge?

**Comment by IntelOrca** [comment]
call it `SpriteTypeToSlowWalkMap`

**Comment by duncanspumpkin** [comment]
Change it to a bool as well since its just a check to see if a sprite type should make it slow walk.

**Comment by IntelOrca** [comment]
Or if you want to be clever and safe the bytes, a single uint64. :smile:

**Comment by janisozaur** [comment]
updated

File: src/peep/peep.c [hunk 1]
```diff
@@ -1291,7 +1296,8 @@ static void set_sprite_type(rct_peep* peep, uint8 type){
 		peep->action = PEEP_ACTION_NONE_2;
 
 	peep->peep_flags &= ~PEEP_FLAGS_SLOW_WALK;
-	if (unk_982134[type] & 1) {
+	assert(type < countof(SpriteTypeToSlowWalkMap));
+	if (SpriteTypeToSlowWalkMap[type] & 1) {
```
Comment (IntelOrca) [inline src/peep/peep.c]: Remove the `& 1`

File: src/peep/peep.c [hunk 2]
```diff
@@ -590,7 +595,7 @@ static void sub_68F41A(rct_peep *peep, int index)
 			peep->action = PEEP_ACTION_NONE_2;
 
 		peep->peep_flags &= ~PEEP_FLAGS_SLOW_WALK;
-		if (unk_982134[sprite_type] & 1) {
+		if (SpriteTypeToSlowWalkMap[sprite_type] & 1) {
```
Comment (IntelOrca) [inline src/peep/peep.c]: Remove the `& 1`

**Comment by janisozaur** [comment]
Done

**Pull request merged**